### PR TITLE
DOC: add searchsorted examples #36411

### PR DESCRIPTION
**Pull request by Mikhaylov-yv**
DOC: add searchsorted examples #36411

### CODE REVIEW

File: pandas/core/base.py [hunk 1]
```diff
@@ -1182,6 +1182,16 @@ def factorize(self, sort: bool = False, na_sentinel: Optional[int] = -1):
...
         array([1, 3])
 
+        >>> ser = pd.Series(pd.to_datetime(['21.10.2020', '22.10.2020', '23.10.2020']))
+        >>> ser
+        0   2020-10-21
+        1   2020-10-22
+        2   2020-10-23
+        dtype: datetime64[ns]
+
+        >>> ser.searchsorted('3/14/2000')
```
Comment (arw2019) [inline pandas/core/base.py:1192]: sorry if i'm being slow but shouldn't this return 0?

**Review by arw2019 — COMMENTED** [review]
Thanks @Mikhaylov-yv for the PR!

A comment

File: pandas/core/base.py (hunk 1 above)
Comment (Mikhaylov-yv) [reply pandas/core/base.py:1192]: You are absolutely right. thanks

Comment (arw2019) [reply pandas/core/base.py:1192]: also need to use the same dates in the constructed series and the output 
(that's why CI is red atm. Once these are fixed I think you're ready to go)

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @Mikhaylov-yv

**Pull request merged**